src/clj/game/cards/ice.clj

```diff
@@ -90,6 +91,7 @@
          new-subs (->> (range total)
                        (reduce (fn [ice _] (add-sub ice sub (:cid ice) args)) card)
                        :subroutines
+                       (map-indexed (fn [idx sub] (assoc sub :index idx)))
```

Comment:
Same as above.

Reply:
Removed this re-indexing as I don't think it should be required for this one since I think it's append only.